File: dashboard/src/api/activity.ts
/**
 * Industrial Wearable AI — Activity timeline API
 */
import { apiClient } from "./client";

export interface TimelineBucket {
  minute: string;
  sewing: number;
  adjusting: number;
  idle: number;
  break: number;
  error: number;
}

export async function getActivityTimeline(
  fromTs: number,
  toTs: number,
  bucketMinutes: number = 1
): Promise<TimelineBucket[]> {
  const { data } = await apiClient.get<TimelineBucket[]>("/api/activity/timeline", {
    params: { from_ts: fromTs, to_ts: toTs, bucket_minutes: bucketMinutes },
  });
  return data;
}

File: backend/app/schemas/activity.py
"""
Industrial Wearable AI — Activity timeline schema for chart API.
"""
from pydantic import BaseModel


class TimelineBucketOut(BaseModel):
    """One time bucket with counts per activity label (minute = "HH:MM")."""

    minute: str
    sewing: int = 0
    adjusting: int = 0
    idle: int = 0
    break_: int = 0  # serialized as "break" in response
    error: int = 0

    model_config = {"populate_by_name": True}
